### ISSUE TITLE
Forthcoming issues with python 3.8.2

### ISSUE THREAD

**Issue by nickjcroucher** [johnlees/PopPUNK#81]
I tried:
`conda create --name pp  -c bioconda python=3.8.2 poppunk`

This results in problems:
```
ImportError: dlopen(/Users/nicholascroucher/miniconda3/envs/pp/lib/python3.8/site-packages/scipy/spatial/qhull.cpython-38-darwin.so, 2): Library not loaded: @rpath/libopenblas.dylib
  Referenced from: /Users/nicholascroucher/miniconda3/envs/pp/lib/python3.8/site-packages/scipy/spatial/qhull.cpython-38-darwin.so
  Reason: image not found
```
Tried to fix with an update of openblas:
```
conda install openblas==0.3.9
...
  hdf5                           1.10.5-nompi_h0cbb7df_1103 --> 1.10.5-nompi_h3e39495_1104
  libblas                                 3.8.0-11_openblas --> 3.8.0-16_openblas
  libcblas                                3.8.0-11_openblas --> 3.8.0-16_openblas
  libgfortran                                       3.0.1-0 --> 4.0.0-2
  liblapack                               3.8.0-11_openblas --> 3.8.0-16_openblas
  libopenblas                              0.3.6-hd44dcd8_6 --> 0.3.9-h3d69b6c_0
  openblas                                 0.3.6-hd44dcd8_6 --> 0.3.9-h2f2564b_0
  scipy               pkgs/main::scipy-1.4.1-py38hebbc057_0 --> conda-forge::scipy-1.4.1-py38h38b60c6_3
```
This generally fixes things, except for a conflict with the specification of pp-sketchlib. So might need a somewhat coordinated update of the conda recipes?

**Comment by johnlees** [comment]
I think this is only a problem on OS X (and is due to sketchlib rather than PopPUNK). 

I ran into a lot of issues with gfortran this week which caused the same v3 vs v4 conflict you have above. It's built with v4 and gets correct dependencies on their build machine, but when you install it locally it's forced to v3 by hdf5. I also saw the openblas issue at some point.

I think I will try statically linking gfortran on OS X when I next update the recipe and see if that solves everything. If not, I will link all the libraries statically.

**Comment by nickjcroucher** [comment]
What a pain - with the openblas update, I can run with mash, didn't realise it was hdf5 causing the problem. I think the main openblas/numpy/gfortran issues seem to be resolved for OSX (maybe that's optimistic)?

**Comment by johnlees** [comment]
I'm just running on Linux for now.

You might find that if you try and run with sketchlib you'll get a similar library not found message for gfortran. You can install `gfortran==4` to a new environment and set:
```
export DYLD_FALLBACK_LIBRARY_PATH=/Users/jlees/miniconda3/envs/gfortran/lib
```
before running.

Obviously a big pain, and I will try and sort it out in the next version

**Comment by johnlees** [comment]
It may just be that this is the preferred dependency: https://github.com/conda-forge/gfortran_impl_osx-64-feedstock

**Comment by johnlees** [comment]
Fixed in pp-sketchlib v1.4.0
